src/app/file-panel.js

```diff
@@ -120,10 +122,15 @@ function filepanel (appAPI, files) {
     this.classList.toggle(css.isVisible)
     this.children[0].classList.toggle('fa-angle-double-right')
     this.children[0].classList.toggle('fa-angle-double-left')
-    api.trigger('ui', [{ type: isHidden ? 'minimize' : 'maximize' }])
+    events.trigger('ui-hidden', [isHidden])
   }
 
   function uploadFile (event) {
+    // TODO This should not go to the file explorer.
+    // The file explorer is merely a view on the current state of
+    // the files module. Please ask the user here if they want to overwrite
+    // a file and then just use `files.add`. The file explorer will
+    // pick that up via the 'fileAdded' event from the files module.
     ;[...this.files].forEach(fileExplorer.api.addFile)
```

Comment:
Please add the `uploadFile` function as an anonymous function directly to the `input` DOM node, so it is much clearer what `this` is. Furthermore, do not call `fileExplorer.api.addFile` but rather `files.addFile` directly, and then the treeView will just pick it up through the fileadded event.